src/lib/djerba/render/database.py

```diff
@@ -0,0 +1,74 @@
+"""Interface with a CouchDB instance for JSON report documents"""
+
+import logging
+import configparser
+import requests
+import json
+import os
+
+import djerba.util.constants as constants
+import djerba.util.ini_fields as ini
+from time import sleep
+from datetime import datetime
+from urllib.parse import urljoin
+from djerba.util.logger import logger
+
+class database(logger):
+    """Class to communicate with CouchDB via the API, eg. using HTTP GET/POST statements"""
+
+    def __init__(self, log_level=logging.WARNING, log_path=None):
+        self.logger = self.get_logger(log_level, __name__, log_path)
+        self.logger.info("Initializing Djerba database object")
+
+    """ Upload json to couchdb"""
+    def upload_file(self, json_path):
+        time = datetime.now()
+        dt_couchDB = time.strftime("%d/%m/%Y %H:%M")
+
+        with open(json_path) as report:
+            data = json.load(report)
+            config = data.get(constants.SUPPLEMENTARY).get(constants.CONFIG)
+            base = config[ini.SETTINGS][ini.ARCHIVE_URL]
+            db = config[ini.SETTINGS][ini.ARCHIVE_NAME]
+            url = urljoin(base, db)
+            report_id = data["report"]["patient_info"]["Report ID"]
+            couch_info = {
+                '_id': '{}'.format(report_id), #DF val auto gen
+                'last_updated': '{}'.format(dt_couchDB),
+            }
+            upload = {**couch_info, **data}
+            headers = {'Content-Type': 'application/json'}
+            submit = requests.post(url= url, headers= headers, json= upload)
+        
+        attempt = 0
+        status = submit.status_code
+        uploaded = False
+        while uploaded == False and attempt < 5:
```

Comment:
There are two possible reasons for failure that interest us:
1. Conflict with existing report
2. Network error/timeout

This code addresses the first one, but not the second, and the loop structure is confusing -- I don't think it will be able to loop more than twice. More comments to follow on Slack.